File: prince-of-versions/src/main/java/co/infinum/princeofversions/Loader.java
package co.infinum.princeofversions;

/**
 * This class loads update resource.
 */
public interface Loader {

    /**
     * Loads update resource into {@link String}.
     *
     * @return Loaded text.
     * @throws Throwable if error happens during load.
     */
    String load() throws Throwable;

}

File: prince-of-versions/src/main/java/co/infinum/princeofversions/Loader.kt
package co.infinum.princeofversions

/**
 * This class loads update resource.
 */
interface Loader {
    /**
     * Loads update resource into [String].
     *
     * @return Loaded text.
     * @throws Throwable if error happens during load.
     */
    @Throws(Throwable::class)
    fun load(): String
}
